languagetool-core/src/main/java/org/languagetool/JLanguageTool.java

```diff
@@ -1772,56 +1806,37 @@ public static void removeTemporaryFiles() {
                     ", text length: " + annotatedText.getPlainText().length());
             //        ", text length: " + annotatedText.getPlainText().length() + ", common word count: " + commonWords.getKnownWordsPerLanguage(annotatedText.getPlainText()));
           }
-          charCount += sentence.length();
-          lineCount += countLineBreaks(sentence);
-
-          // calculate matching column:
-          int lineBreakPos = sentence.lastIndexOf('\n');
-          if (lineBreakPos == -1) {
-            columnCount += sentence.length();
-          } else {
-            if (lineBreakPos == 0) {
-              columnCount = sentence.length();
-              if (!language.getSentenceTokenizer().singleLineBreaksMarksPara()) {
-                columnCount--;
-              }
-            } else {
-              columnCount = sentence.length() - lineBreakPos;
-            }
-          }
         } catch (ErrorRateTooHighException e) {
           throw e;
         } catch (Exception e) {
           throw new RuntimeException("Could not check sentence (language: " + language + "): <sentcontent>"
-                  + StringUtils.abbreviate(analyzedSentence.toTextString(), 500) + "</sentcontent>", e);
+                  + StringUtils.abbreviate(sentence.analyzed.toTextString(), 500) + "</sentcontent>", e);
         }
       }
       return ruleMatches;
     }
 
-    private LineColumnRange getLineColumnRange(RuleMatch match) {
-      LineColumnPosition fromPos = new LineColumnPosition(-1, -1);
-      LineColumnPosition toPos = new LineColumnPosition(-1, -1);
-      LineColumnPosition pos = new LineColumnPosition(0, 0);
-      int charCount = 0;
-      for (AnalyzedSentence analyzedSentence : analyzedSentences) {
-        for (AnalyzedTokenReadings readings : analyzedSentence.getTokens()) {
-          String token = readings.getToken();
-          if ("\n".equals(token)) {
-            pos.line++;
-            pos.column = 0;
-          }
-          pos.column += token.length();
-          charCount += token.length();
-          if (charCount == match.getFromPos()) {
-            fromPos = new LineColumnPosition(pos.line, pos.column);
-          }
-          if (charCount == match.getToPos()) {
-            toPos = new LineColumnPosition(pos.line, pos.column);
-          }
-        }
+    private LineColumnPosition findLineColumn(int offset) {
+      if (sentences.isEmpty()) return new LineColumnPosition(0, 0);
+
+      SentenceData sentence = findSentenceContaining(offset);
+      String prefix = sentence.text.substring(0, offset - sentence.startOffset);
+      return new LineColumnPosition(
+        sentence.startLine + countLineBreaks(prefix),
+        processColumnChange(sentence.startColumn, prefix));
+    }
+
+    private SentenceData findSentenceContaining(int offset) {
+      int low = 0;
+      int high = sentences.size() - 1;
+      while (low <= high) {
+        int mid = (low + high) >>> 1;
```

Comment:
This operator is not very common, could you add a comment?

Reply:
It's inspired by JDK code, but indeed, this could be simplified. Will do!